src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -0,0 +1,722 @@
+
+#include <memory>
+
+
+#include "infer_request.hpp"
+#include "async_infer_request.hpp"
+
+// #include "perf_counter.hpp"
+// #include "graph_debug_dump.hpp"
+
+
+#include "openvino/runtime/exec_model_info.hpp"
+#include "openvino/runtime/properties.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/util/common_util.hpp"
+#include "transformations/rt_info/fused_names_attribute.hpp"
+#include "transformations/utils/utils.hpp"
+#include "xml_parse_utils.h"
+
+#include "ie_ngraph_utils.hpp"
+#include "ie_plugin_config.hpp"
+#include "ie_algorithm.hpp"
+
+#include "converter_utils.hpp"
+
+template <typename T>
+using NodeMap = std::unordered_map<ngraph::Node*, T>;
+
+ov::hetero::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
+                                         const std::shared_ptr<const ov::IPlugin>& plugin,
+                                         const Configuration& cfg,
+                                         bool loaded_from_cache)
+    : ov::ICompiledModel(model, plugin),
+      m_cfg(cfg),
+      m_model(model),
+      m_name(model->get_name()),
+      m_loaded_from_cache(loaded_from_cache) {
+    try {
+        bool dumpDotFile = false;
+        if (std::getenv("OPENVINO_HETERO_VISUALIZE")) {
+            dumpDotFile = true;
+        } else {
+            dumpDotFile = m_cfg.dump_graph;
+        }
+
+        ov::SupportedOpsMap queryNetworkResult;
+        auto orderedOps = m_model->get_ordered_ops();
+
+        bool allEmpty = true;
+        // Get user defined affinity
+        for (auto&& node : orderedOps) {
+            auto& nodeInfo = node->get_rt_info();
+            auto itInfo = nodeInfo.find("affinity");
+            if (itInfo != nodeInfo.end()) {
+                OPENVINO_ASSERT(itInfo->second.is<std::string>());
+                queryNetworkResult.emplace(node->get_friendly_name(), itInfo->second.as<std::string>());
+                allEmpty = false;
+            }
+        }
+
+        if (queryNetworkResult.empty())
+            queryNetworkResult = plugin->query_model(model, m_cfg.GetHeteroConfig());
+
+        using Input = ov::Input<ov::Node>;
+        using NodeSet = std::unordered_set<ov::Node*>;
+        using InputSet = std::set<Input>;
+
+        auto InputNode = [](const Input& input) {
+            return input.get_source_output().get_node();
+        };
+
+        std::unordered_set<std::string> devices;
+        NodeMap<std::string> affinities;
+        // Check that all nodes has user or plugin defined affinities
+        for (auto&& node : orderedOps) {
+            auto itAffinity = queryNetworkResult.find(node->get_friendly_name());
+            if (itAffinity != queryNetworkResult.end()) {
+                affinities[node.get()] = itAffinity->second;
+                devices.emplace(itAffinity->second);
+            } else if (allEmpty) {
+                IE_THROW() << "Hetero device used default fallback policy, but some layers eg: \n(Name:"
```

Comment:
Please throw only new exceptions
```suggestion
                OPENVINO_THROW() 
```

Reply:
Done